doc/htm.md

```diff
@@ -0,0 +1,37 @@
+HTM Indexing
+============
+
+Overview                        {#htm-overview}
+========
```

Comment:
This section title uses the same markdown level as document title above, would it be better to give it different level (e.g. ----------)?